Applications/Spire/Source/Ui/IntegerInputWidget.cpp

```diff
@@ -1,89 +1,50 @@
 #include "Spire/Ui/IntegerInputWidget.hpp"
-#include <algorithm>
-#include <QKeyEvent>
-#include <QLineEdit>
-#include <QMouseEvent>
-#include "Spire/Spire/Dimensions.hpp"
 
 using namespace boost::signals2;
 using namespace Spire;
 
-namespace {
-  auto BUTTON_X() {
-    static auto x = scale_width(12);
-    return x;
-  }
+IntegerInputWidget::IntegerInputWidget(std::int64_t value, QWidget* parent)
+    : QWidget(parent) {
+  m_input_widget = new NumericInputWidget(value, this);
+  m_input_widget->set_decimals(0);
+  m_input_widget->connect_change_signal([=] (auto value) {
+    m_change_signal(value.extract_unsigned_long_long());
```

Comment:
There is almost never a good reason to operate on unsigned values. It's very rare that you will use unsigned values except when you actually need some kind of overflow behavior.